porcupine/plugins/filemanager.py

```diff
@@ -122,6 +157,31 @@ def rename(old_path: Path) -> None:
         tab.path = new_path / tab.path.relative_to(old_path)
 
 
+def paste(new_path: Path) -> None:
+    assert copy_path is not None
+    new_file_path = new_path / copy_path.name
+
+    if new_file_path.exists():
+        if new_file_path.parent == copy_path.parent:  # copying file in the same dir (sequential number copy)
+            path_number = 0
+            path_stem = new_file_path.stem
+            path_suffix = new_file_path.suffix
+            while new_file_path.exists():
+                path_number += 1
+                new_file_path = new_file_path.with_name(path_stem + f" ({path_number})" + path_suffix)
+        else:
+            new_file_path = ask_name_for_renaming(new_file_path, is_paste=True)  # rename-overwrite file conflict
+
+    if new_file_path is not None:
+        new_file_path.touch()
```

Comment:
This shouldn't be necessary, `shutil.copy` creates the file if it doesn't exist.